tst/EnergyPlus/unit/AirflowNetworkSolver.unit.cc

```diff
@@ -259,3 +259,96 @@ TEST_F(EnergyPlusFixture, AirflowNetwork_SolverTest_GenericCrack)
     EXPECT_DOUBLE_EQ(0.000065 * std::pow(10.0, 0.65), DF[0]);
     EXPECT_EQ(0.0, DF[1]);
 }
+
+TEST_F(EnergyPlusFixture, AirflowNetwork_SolverTest_SpecifiedMassFlow)
+{
+
+    int NF;
+    std::array<Real64, 2> F = {0.0, 0.0};
+    std::array<Real64, 2> DF = {0.0, 0.0};
+
+    AirflowNetwork::SpecifiedMassFlow element;
+    element.mass_flow = 0.1;
+
+    AirflowNetwork::AirProperties state0, state1;
+
+    Real64 dp{10.0};
+    Real64 f = element.mass_flow;
+
+    // Linear
+    NF = element.calculate(*state, true, dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+
+    NF = element.calculate(*state, true, -dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+
+    // Nonlinear tests
+    NF = element.calculate(*state, false, dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+
+    NF = element.calculate(*state, false, -dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+}
+
+TEST_F(EnergyPlusFixture, AirflowNetwork_SolverTest_SpecifiedVolumeFlow)
+{
+
+    int NF;
+    std::array<Real64, 2> F = {0.0, 0.0};
+    std::array<Real64, 2> DF = {0.0, 0.0};
+
+    AirflowNetwork::SpecifiedVolumeFlow element;
+    element.volume_flow = 0.1;
+
+    AirflowNetwork::AirProperties state0, state1;
+    Real64 density = state0.density; // = state1.density
+
+    Real64 dp{10.0};
+    Real64 f = element.volume_flow * density;
+
+    // Linear
+    NF = element.calculate(*state, true, dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+
+    NF = element.calculate(*state, true, -dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+
+    // Nonlinear tests
+    NF = element.calculate(*state, false, dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+
+    NF = element.calculate(*state, false, -dp, 0, 1.0, 1.0, state0, state1, F, DF);
+    EXPECT_EQ(f, F[0]);
+    EXPECT_EQ(0.0, F[1]);
+    EXPECT_EQ(0.0, DF[0]);
+    EXPECT_EQ(0.0, DF[1]);
+    EXPECT_EQ(1, NF);
+}
```

Comment:
And finally, a couple of unit tests that verify we always get back the same flow.